electron_app/electron.js

```diff
@@ -109,34 +109,51 @@ Main._initializeApplication = function () {
     electron.ipcMain.on('app_ready', async(event) => {
       autoUpdater.autoDownload = false;
 
-      autoUpdater.checkForUpdates();
+      autoUpdater.checkForUpdates().then((result) => {
+        console.log(result.updateInfo.version);
 
-      const currentVersion = electron.app.getVersion();
-      const currentVersionIsPrerelease = prereleaseRegex.test(currentVersion);
-      autoUpdater.allowPrerelease = currentVersionIsPrerelease;
 
-      const downloadCancellationToken = new CancellationToken();
+        const currentVersion = electron.app.getVersion();
+        const currentVersionIsPrerelease = prereleaseRegex.test(currentVersion);
+        autoUpdater.allowPrerelease = currentVersionIsPrerelease;
 
-      console.log(`CurrentVersion: ${currentVersion}, CurrentVersionIsPrerelease: ${currentVersionIsPrerelease}`);
+        const downloadCancellationToken = new CancellationToken();
 
-      autoUpdater.addListener('error', (error) => {
-        event.sender.send('update_error');
-      });
+        console.log(`CurrentVersion: ${currentVersion}, CurrentVersionIsPrerelease: ${currentVersionIsPrerelease}`);
+
+        autoUpdater.addListener('error', (error) => {
+          event.sender.send('update_error');
+        });
+
+        autoUpdater.addListener('update-available', (info) => {
+          event.sender.send('update_available', result.updateInfo.version);
 
-      autoUpdater.addListener('update-available', (info) => {
-        event.sender.send('update_available');
+          electron.ipcMain.on('download_update', (event) => {
+            autoUpdater.downloadUpdate(downloadCancellationToken);
+          });
 
-        electron.ipcMain.on('download_update', (event) => {
-          autoUpdater.downloadUpdate(downloadCancellationToken);
+          electron.ipcMain.on('show_release_notes', () => {
+            const releaseNotesWindow = new electron.BrowserWindow({
+              width: 600,
+              height: 600,
+              title: `Release Notes ${result.updateInfo.version}`,
+              minWidth: 600,
+              minHeight: 600,
+            })
+
+            releaseNotesWindow.loadURL(`https://github.com/process-engine/bpmn-studio/releases/tag/v${result.updateInfo.version}`);
+          });
         });
-      });
 
-      autoUpdater.addListener('update-downloaded', (info) => {
-        event.sender.send('update_downloaded');
+        autoUpdater.addListener('update-downloaded', (info) => {
+          event.sender.send('update_downloaded');
 
-        electron.ipcMain.on('quit_and_install', (event) => {
-          autoUpdater.quitAndInstall();
+          electron.ipcMain.on('quit_and_install', (event) => {
```

Comment:
Like above, `event` shadows a previously declared variable, but is kinda redundant here anyway.

```suggestion
          electron.ipcMain.on('quit_and_install', () => {
```